### ISSUE TITLE
`{ingress,sidecar} -> ztunnel'd app` should use HBONE

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#43587]
Currently we only detect if the dest has a waypoint from Envoy clients AFAIK

**Comment by linsun** [comment]
Been wondering about this... do you think this is also why Kiali dashboard didn't show mtls from ingress gateway to productpage?

**Comment by howardjohn** [comment]
yes

**Comment by howardjohn** [comment]
https://github.com/istio/istio/issues/43586 for prometheus->productpage not showing it as well

**Comment by linsun** [comment]
cool, thanks! was going to ping you about it :)

**Comment by linsun** [comment]
Looking at ztunnel config in my env, I noticed the nativeHbone is false for ingress gateway. Shouldn't this be true?

```
    "10.244.2.2": {
      "workloadIp": "10.244.2.2",
      "waypointAddresses": [],
      "gatewayAddress": null,
      "protocol": "TCP",
      "name": "istio-ingressgateway-589bcdf5d6-97r46",
      "namespace": "istio-system",
      "serviceAccount": "istio-ingressgateway-service-account",
      "workloadName": "istio-ingressgateway",
      "workloadType": "deployment",
      "canonicalName": "istio-ingressgateway",
      "canonicalRevision": "latest",
      "node": "ambient-worker2",
      "nativeHbone": false,
      "authorizationPolicies": [],
      "status": "Healthy"
    },
```

**Comment by howardjohn** [comment]
actually, no. Ingress does NOT accept HBONE traffic currently. Same reason it is not labeled for auto-mtls (except gateway can be manually put into mtls-mode with ISTIO_MUTUAL; we do not have an HBONE mode at all).

**Comment by danehans** [comment]
The L4 authz policy in the latest ambient [getting started guide](https://preliminary.istio.io/latest/docs/ops/ambient/getting-started/) is failing due to this bug. According to the docs, the sleep pod should be able to curl the productpage pod through the ingress gateway but I get:
```
kubectl exec deploy/sleep -- curl -s http://istio-ingressgateway.istio-system/productpage | head -n10
upstream connect error or disconnect/reset before headers. reset reason: connection termination
```

Should the guide be updated until this bug is fixed?

**Comment by howardjohn** [comment]
@danehans I cannot reproduce that

**Comment by linsun** [comment]
> actually, no. Ingress does NOT accept HBONE traffic currently. Same reason it is not labeled for auto-mtls (except gateway can be manually put into mtls-mode with ISTIO_MUTUAL; we do not have an HBONE mode at all).

Thanks for the clarification!  So  ingress gateway can only send outbound traffic via HBONE?

**Comment by howardjohn** [comment]
> > actually, no. Ingress does NOT accept HBONE traffic currently. Same reason it is not labeled for auto-mtls (except gateway can be manually put into mtls-mode with ISTIO_MUTUAL; we do not have an HBONE mode at all).
> 
> Thanks for the clarification! So ingress gateway can only send outbound traffic via HBONE?

Yes, for now anyhow

**Comment by linsun** [comment]
> The L4 authz policy in the latest ambient [getting started guide](https://preliminary.istio.io/latest/docs/ops/ambient/getting-started/) is failing due to this bug. According to the docs, the sleep pod should be able to curl the productpage pod through the ingress gateway but I get:
> 
> ```
> kubectl exec deploy/sleep -- curl -s http://istio-ingressgateway.istio-system/productpage | head -n10
> upstream connect error or disconnect/reset before headers. reset reason: connection termination
> ```
> 
> Should the guide be updated until this bug is fixed?

This is the same error I got on the same step.

**Comment by howardjohn** [comment]
ok let me try again

**Comment by howardjohn** [comment]
I still cannot reproduce..
```
istio-system/ztunnel-gbkjm[istio-proxy]: 2023-03-01T17:05:31.216307Z  INFO outbound{id=09afb06540d0cf71cd85021bf11e7a16 source_ip=sleep-bc9998558-hrgx6}: ztunnel::proxy::outbound: Proxying to istio-ingressgateway-5b854cb49d-pplc2:8080 using TCP via istio-ingressgateway-5b854cb49d-pplc2:8080 type Direct
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.223196Z  INFO inbound{id=2f29490145c5acb1e6fbf0f4bd33562c peer_ip=istio-ingressgateway-5b854cb49d-pplc2 peer_id=spiffe://cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account}: ztunnel::proxy::inbound: got CONNECT request to productpage-v1-58b4c9bff8-nw59v:9080
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.227712Z  INFO outbound{id=5611712584c8e9751805ce461d62d552 source_ip=productpage-v1-58b4c9bff8-nw59v}: ztunnel::proxy::outbound: proxying to details-v1-6997d94bb9-glt9b:9080 using node local fast path
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.227967Z  INFO outbound{id=5611712584c8e9751805ce461d62d552 source_ip=productpage-v1-58b4c9bff8-nw59v}: ztunnel::proxy::outbound: complete dur=428.86µs
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.233119Z  INFO outbound{id=3d60744cfe426684e373313c56382525 source_ip=productpage-v1-58b4c9bff8-nw59v}: ztunnel::proxy::outbound: proxying to reviews-v3-589cb4d56c-x8lrh:9080 using node local fast path
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.233495Z  INFO outbound{id=3d60744cfe426684e373313c56382525 source_ip=productpage-v1-58b4c9bff8-nw59v}: ztunnel::proxy::outbound: complete dur=680.42µs
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.242254Z  INFO outbound{id=56253aff63222d99418b95d142142211 source_ip=reviews-v3-589cb4d56c-x8lrh}: ztunnel::proxy::outbound: complete dur=8.477215091s
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.242730Z  INFO outbound{id=c91c0bd51c7aaab266e83da47cc7bd83 source_ip=reviews-v3-589cb4d56c-x8lrh}: ztunnel::proxy::outbound: proxying to ratings-v1-b8f8fcf49-6p992:9080 using node local fast path
istio-system/ztunnel-qcmm5[istio-proxy]: 2023-03-01T17:05:31.243053Z  INFO outbound{id=c91c0bd51c7aaab266e83da47cc7bd83 source_ip=reviews-v3-589cb4d56c-x8lrh}: ztunnel::proxy::outbound: complete dur=540.16µs
istio-system/ztunnel-gbkjm[istio-proxy]: 2023-03-01T17:05:31.262765Z  INFO outbound{id=09afb06540d0cf71cd85021bf11e7a16 source_ip=sleep-bc9998558-hrgx6}: ztunnel::proxy::outbound: complete dur=46.921509ms
```

**Comment by danehans** [comment]
I can no longer reproduce https://github.com/istio/istio/issues/43587#issuecomment-1450456467.

**Comment by danehans** [comment]
It appears that a Gateway is also affected by this bug.

To reproduce, create a Gateway in a namespace where bookinfo is running:
```
$ kubectl apply -f samples/bookinfo/gateway-api/bookinfo-gateway.yaml
```
 
Test connectivity through the Gateway:
```
$ kubectl exec deploy/sleep -- sh -c "for i in \$(seq 1 10); do curl -s http://bookinfo-gateway-istio.default/productpage | grep reviews-v.-; done"
        <u>reviews-v2-65c4dc6fdc-qcpg6</u>
        <u>reviews-v2-65c4dc6fdc-qcpg6</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v2-65c4dc6fdc-qcpg6</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
```

Enable ambient for the namespace:
```
$ kubectl label namespace default istio.io/dataplane-mode=ambient
```

Verify HBONE is set for the Gateway:
```
$ istioctl pc w ds/ztunnel -n istio-system
NAME                                    NAMESPACE          IP          NODE                  L7 SUPPORT PROTOCOL
bookinfo-gateway-istio-6f6fb947df-svwtn default            10.244.2.10 ambient-worker        false      HBONE
productpage-v1-8b588bf6d-hv26t          default            10.244.2.9  ambient-worker        false      HBONE
...
```

Retest connectivity through the Gateway:
```
$ kubectl exec deploy/sleep -- sh -c "for i in \$(seq 1 10); do curl -s http://bookinfo-gateway-istio.default/productpage | grep reviews-v.-; done"
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v2-65c4dc6fdc-qcpg6</u>
command terminated with exit code 1
```

Remove the Gateway from ambient:
```
$ export GATEWAY_POD=$(kubectl get pods -l istio.io/gateway-name=bookinfo-gateway -o jsonpath='{.items[*].metadata.name}')
$ kubectl annotate po/"$GATEWAY_POD" ambient.istio.io/redirection=disabled --overwrite
```

Retest connectivity:
```
kubectl exec deploy/sleep -- sh -c "for i in \$(seq 1 10); do curl -s http://bookinfo-gateway-istio.default/productpage | grep reviews-v.-; done"
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
        <u>reviews-v2-65c4dc6fdc-qcpg6</u>
        <u>reviews-v3-c9c4fb987-46thc</u>
        <u>reviews-v2-65c4dc6fdc-qcpg6</u>
        <u>reviews-v1-569db879f5-q9jmj</u>
```
